recipe/meta.yaml

```
@@ -13,57 +14,131 @@ source:
-    - snappy
+outputs:
+  # Shared library, considered unstable!
+  # See: https://github.com/facebook/folly/#build-notes
+  - name: folly
+    script: build_folly.sh   # [unix]
+    script: build_folly.bat  # [win]
+    version: {{ version }}
+    build:
+      number: {{ build_number }}
+      string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}{{ build_ext }}
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+      host:
+        - boost-cpp
+        - double-conversion
+        - fmt {{ fmt }}
+        - gflags
+        - glog
+        - jemalloc        # [(not osx) and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
+        - jemalloc-local  # [osx and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
+        - libaio  # [linux]
+        - libevent
+        - libsodium
+        - snappy
+        - bzip2
+        - lz4-c
+        - openssl
+        - xz
+        - zlib
+        - zstd
+      run:
+        - boost-cpp
+        - jemalloc        # [(not osx) and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
+        - jemalloc-local  # [osx and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
+        - snappy
+    test:
+      requires:
+        - {{ compiler('cxx') }}
+      commands:
+        - test -f ${PREFIX}/lib/libfolly${SHLIB_EXT}              # [unix]
+        - test -f ${PREFIX}/include/folly/folly-config.h          # [unix]
+        - $CXX $CXXFLAGS $LDFLAGS  /home/conda/recipe_root/arena_test.c -lfolly -o arena_test  # [unix]
+        - test -f ./arena_test  # [unix]
+        - ./arena_test  # [unix]
+
+    about:
+      home: https://github.com/facebook/folly
+      license: Apache-2.0
+      license_file: LICENSE
+      summary: Facebook Open-source Library — Shared Library
+      description: Shared Library for folly, considered unstable!
+      doc_url: https://github.com/facebook/folly/tree/master/folly/docs
+      dev_url: https://github.com/facebook/folly
+
+  # Static library
+  - name: folly-static
+    script: build_folly.sh   # [unix]
+    script: build_folly.bat  # [win]
+    version: {{ version }}
+    build:
+      number: {{ build_number }}
+      string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}{{ build_ext }}
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+      host:
+        - boost-cpp
+        - double-conversion
+        - fmt {{ fmt }}
+        - gflags
+        - glog
+        - jemalloc        # [(not osx) and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
+        - jemalloc-local  # [osx and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
+        - libaio  # [linux]
+        - libevent
+        - libsodium
+        - snappy
+        - bzip2
+        - lz4-c
+        - openssl
+        - xz
+        - zlib
+        - zstd
```

Please deduplicate the headers, they should only be in the shared build.
```suggestion
        - zstd
        - {{ pin_subpackage('folly', exact=True) }}
```